lua/core/clock.lua

```diff
@@ -210,6 +218,9 @@ function clock.add_params()
       if source == "internal" then clock.internal.set_tempo(bpm)
       elseif source == "link" then clock.link.set_tempo(bpm) end
       norns.state.clock.tempo = bpm
+      if clock.tempo_change_handler ~= nil then
+        clock.tempo_change_handler(bpm, source)
```

Comment:
What's the reason for passing source as an argument here? It could be easily retrieved as a param value if necessary, could it?

Reply:
for sure! i think i might actually have subconsciously been angling for adding a `clock.source_change_handler` as well -- would that be too much, or would it be a good completion of the callback set?

Reply:
> i do think i might actually have subconsciously been angling for a clock.source_change_handler as well -- would that be too much to add, or would it be a good completion of the callback set?

yep, i've thought about it too, but can't imagine a use-case for the source-changed kind of callback at the moment, as the clock API is largely source- and tempo-agnostic by design anyway.

so i'd suggest sticking with a more minimalist approach here and leaving the tempo-changed callback having only the bpm parameter, and if a source-changed callback is necessary in the future - add it later at some point, wdyt?